### ISSUE TITLE
ECS Dashboard

### ISSUE THREAD

**Issue by erlend-sh** [bevyengine/bevy#446]
Flecs by @SanderMertens has a beautiful dashboard: https://github.com/flecs-hub/flecs-dash ([previous iteration here](https://github.com/flecs-hub/flecs-systems-admin )).

![](https://images-ext-2.discordapp.net/external/rCZY_brkz3j9FnaMUfjRmPcxLXXob7VrT1Vd-z7wRuk/https/media.discordapp.net/attachments/750833446355730482/750834735604826225/Screen_Shot_2020-08-31_at_5.59.49_PM.png)

This could be adapted to work with Bevy ECS.

> *Sander:*
> I'd be happy to help 🙂 note that the project is still very much a work in progress
> I imagine it wouldn't be too difficult to port as I think bevy & flecs work similarly under the hood (both archetypes), and all you need is the right REST interface to make the dashboard work

**Comment by smokku** [comment]
Isn't that against:
> [...] the editor should be built in the engine. Godot uses this approach and it is so smart. Doing so dogfoods the engine's UI system and creates positive feedback loops. Improvements to the editor are also often improvements to the core engine. It also makes sure your engine is flexible enough to build tooling (and not just games). I personally consider building an engine's editor in another stack to be a missed opportunity (ex: the web, QT, native widgets).

**Comment by chemicstry** [comment]
Yes, but the editor is still far away. I think this is more a suggestion for a 3rd party crate, which could be used to debug ECS

**Comment by cart** [comment]
Yeah this is a good idea, but integration should be a 3rd party plugin. The editor will (hopefully) eventually have dashboard-like functionality so thats what we would merge into this repo.

**Comment by fopsdev** [comment]
This is great. As well it could help dogfooding bevys own network implementation (hmmm not even sure if there is one) :)

**Comment by memoryruins** [comment]
@fopsdev currently networking is being prototyped and experimented with in external plugins https://github.com/bevyengine/awesome-bevy#networking

**Comment by Affinator** [comment]
Short reference to the Bevy Remote Protocol: https://github.com/bevyengine/bevy/pull/14880

**Back to the issue:**
IMHO this would be a great addition to working in the editor, which i think is more asset focussed, to allow profiling at runtime. This could of course also be integrated into the editor, but I for one still prefere doing everything in code and would love to have a simple profiling protocol to hook into.

**Comment by alice-i-cecile** [comment]
I think this is ultimately duplicative of #15636. Going to consolidate this into that issue, as we have a clearer idea of what we'd like to build now :)
